pkg/context/detached.go

```diff
@@ -0,0 +1,35 @@
+package context
+
+import (
+	"context"
+	"time"
+)
+
+// Detached enables creating a copy of a context with all of its values,
+// but cancelling the parent context will not cancel the detached.
+// Used when forwarding a request context to a goroutine, and when the request
+// context is cancelled the goroutine can continue its operation.
+type Detached struct {
+	ctx context.Context
+}
+
+// NewDetached returns a context that is never canceled.
```

Comment:
no so true. the purpose is not that it never cancelled but perhaps it is detached from the parent ctx, meaning, not affected by its parent